### PR TITLE
Migrate bls into lodestar monorepo

### LINKED ISSUE #291: Add bls to lodestar repository
Integrate https://github.com/chainsafe/bls-js as lerna package in lodestar repository.
See how we do it in ssz first

### PR DESCRIPTION
**Pull request by wemeetagain**
Move bls repo with history
Minimal changes
  - tests/ ->test/
  - tsconfig/babelrc updates

resolves #291

### CODE REVIEW

File: packages/bls/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+### ChainSafe ETH2.0 Projects
+Note:
+There is a lot of work being done that are core infrastructural pieces for Eth2.0. Contributions to any of the below repositories would be greatly appreciated. All the libraries are written in TypeScript (or in the process of being converted from pure JS to TypeScript):
+<br />
+-- [PM / Meta Repo](https://github.com/ChainSafe/Sharding)<br />
+|-- [Beacon Chain](https://github.com/ChainSafe/lodestar_chain)<br />
+|-- [Simple Serialize (SSZ)](https://github.com/ChainSafe/ssz-js)<br />
+|-- [Fixed Size Numbers](https://github.com/ChainSafe/fixed-sized-numbers-ts/)<br />
+|-- [BLS Signatures and Signature Aggregation](https://github.com/ChainSafe/bls-js)<br />
```
Comment (mpetrunic) [inline packages/bls/README.md]: update links

File: packages/bls/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# bls-js
+
+[![Build Status](https://travis-ci.org/ChainSafe/bls-js.svg?branch=master)](https://travis-ci.org/ChainSafe/bls-js)
+[![codecov](https://codecov.io/gh/ChainSafe/bls-js/branch/master/graph/badge.svg)](https://codecov.io/gh/ChainSafe/bls-js)
+[![](https://badges.gitter.im/chainsafe/lodestar.svg)](https://gitter.im/chainsafe/lodestar?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+![ETH2.0_Spec_Version 0.8.0](https://img.shields.io/badge/ETH2.0_Spec_Version-0.8.0-2e86c1.svg)
```
Comment (mpetrunic) [inline packages/bls/README.md:5]: update badges

File: packages/bls/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+>[spec](https://github.com/ethereum/eth2.0-specs/blob/master/specs/bls_signature.md)
+
+>[test vectors](https://github.com/ethereum/eth2.0-spec-tests/tree/master/tests/bls)
+
+## Usage
+- `yarn add @chainsafe/bls`
+
+## Development
+- `git clone --recursive git@github.com:ChainSafe/bls-js.git`
```
Comment (mpetrunic) [inline packages/bls/README.md]: ^^

File: packages/bls/package.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
+{
+  "name": "@chainsafe/bls",
+  "version": "0.1.6",
+  "description": "Implementation of bls signature verification for ethereum 2.0",
+  "main": "lib/index.js",
+  "types": "lib/index.d.ts",
+  "homepage": "https://github.com/chainsafe/bls-js",
```
Comment (mpetrunic) [inline packages/bls/package.json]: update

File: packages/bls/package.json [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  "author": "ChainSafe Systems",
+  "license": "Apache-2.0",
```
Comment (mpetrunic) [inline packages/bls/package.json:9]: do we have to have that license because of apache milagro?

File: packages/bls/package.json [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    "build-web": "webpack --mode production --entry ./lib/web.js --output ./dist/bls.min.js",
+    "check-types": "tsc --noEmit",
+    "lint": "eslint --ext .ts src/",
+    "lint-fix": "eslint --ext .ts src/ --fix",
+    "pretest": "yarn check-types",
+    "prepublishOnly": "yarn build",
+    "test:unit": "mocha -r ./.babel-register \"test/unit/**/*.test.ts\"",
+    "test:spec": "mocha -r ./.babel-register \"test/spec/**/*.test.ts\"",
+    "test": "nyc -r lcov -e .ts -x \"*.test.ts\" mocha -r ./.babel-register \"test/**/*.test.ts\" && nyc report",
+    "coverage": "codecov"
```
Comment (mpetrunic) [inline packages/bls/package.json]: missing flag bls, so you can have badge and coverage dedicated to this project

File: packages/bls/package.json [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    "build-types": "tsc --declaration --emitDeclarationOnly",
+    "build-web": "webpack --mode production --entry ./lib/web.js --output ./dist/bls.min.js",
+    "check-types": "tsc --noEmit",
+    "lint": "eslint --ext .ts src/",
+    "lint-fix": "eslint --ext .ts src/ --fix",
+    "pretest": "yarn check-types",
+    "prepublishOnly": "yarn build",
+    "test:unit": "mocha -r ./.babel-register \"test/unit/**/*.test.ts\"",
+    "test:spec": "mocha -r ./.babel-register \"test/spec/**/*.test.ts\"",
+    "test": "nyc -r lcov -e .ts -x \"*.test.ts\" mocha -r ./.babel-register \"test/**/*.test.ts\" && nyc report",
```
Comment (mpetrunic) [inline packages/bls/package.json]: you should move coverage to test:unit as this never gets called

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: packages/bls/package.json (hunk 5 above)
Comment (wemeetagain) [reply packages/bls/package.json:9]: I don't think so, I just left it as is.

**Comment by wemeetagain** [comment]
@mpetrunic kinda stumped at the current CI error.
It looks like a dependency order issue?
starting to build eth2.0-ssz-types before finishing building ssz?
I wonder if its due to the current circular dependency between ssz and eth2.0-ssz-types?

**Comment by mpetrunic** [comment]
@wemeetagain I think the reason is invalid tsconfig. Look at build directory, type declarations end up in /build/src/*.d.ts.

We need to add to each project rootDir and includes

**Comment by wemeetagain** [comment]
@mpetrunic its strange because the tests have been working before, even with ssz added. And the CI error refered to ssz, seemingly unrelated to bls declaration files.

**Comment by wemeetagain** [comment]
omg! finally green :money_mouth_face:

**Review by mpetrunic — APPROVED** [review]

**Review by GregTheGreek — APPROVED** [review]

**Pull request merged**